### PR TITLE
Psr 29 echo utility improvements

### PR DESCRIPTION
**Pull request by abstractmachines**
Improvements:
- Naming conventions (explicit + verbose)
- Container option indexing (safe return values)
- Whitespace

### CODE REVIEW

File: Echo/ApplicationInputHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+                    _getOptHelpFlag = true;
+                }
+
+                else
+                {
+                    _userFileName = optarg;
+
+                    _echoMessageToFileFlag = true;
+
```
Comment (dmiller-polysync) [inline Echo/ApplicationInputHandler.cpp]: Extra whitespace.

File: Echo/Echo.cpp [hunk 2]
```diff
@@ -98,12 +55,27 @@ class PolySyncEcho : public polysync::Node
...
 
+    // Nodes will only connect if help option -h not used,
+    // and, if all arguments are valid.
+    if ( echo.optionsParse( argc, argv ) )
+    {
+        if ( echo.wasHelpRequested() )
+        {
+            echo.printEchoHelpPage();
+        }
+
```
Comment (dmiller-polysync) [inline Echo/Echo.cpp]: extra whitespace

File: Echo/EchoHelp.cpp [hunk 3]
```diff
@@ -0,0 +1,85 @@
+#include "PolySyncDataModel.hpp"
+#include "EchoHelp.hpp"
+
+using namespace std;
+
+namespace polysync
+{
+
+EchoHelp::EchoHelp()
```
Comment (dmiller-polysync) [inline Echo/EchoHelp.cpp]: Probably don't need to define or prototype a constructor if it doesn't do anything.

Comment (abstractmachines) [inline Echo/EchoHelp.cpp]: Addressed here: https://github.com/PolySync/PolySync-CPP-Examples/pull/8/commits/6cf3c271fe519a152decdf4316b833922f510df9

File: Echo/Echo.cpp (hunk 2 above)
Comment (abstractmachines) [inline Echo/Echo.cpp]: Addressed here: https://github.com/PolySync/PolySync-CPP-Examples/pull/8/commits/6cf3c271fe519a152decdf4316b833922f510df9

File: Echo/ApplicationInputHandler.cpp (hunk 1 above)
Comment (abstractmachines) [inline Echo/ApplicationInputHandler.cpp]: addressed here: https://github.com/PolySync/PolySync-CPP-Examples/pull/8/commits/11a1c228886fcf7bbe8f3ad9687e2a3b6dd0963b

File: Echo/ApplicationInputHandler.cpp [hunk 4]
```diff
@@ -0,0 +1,183 @@
+#include "PolySyncDataModel.hpp"
+#include "ApplicationInputHandler.hpp"
+#include "getopt.h"
+#include <algorithm>
```
Comment (CrockAgile) [inline Echo/ApplicationInputHandler.cpp:1]: Small style thing. Should library includes like `<algorithm>` be before the polysync / example includes?

File: Echo/Echo.cpp [hunk 5]
```diff
@@ -35,59 +36,13 @@
  *
  */
 
-#include <iostream>
-#include <PolySyncNode.hpp>
-#include <PolySyncDataModel.hpp>
+#include <EchoNode.hpp>
```
Comment (CrockAgile) [inline Echo/Echo.cpp]: Is EchoNode.hpp a system installed header or within this example project? If it is for this example, then maybe should be `#include "EchoNode.hpp"`

File: Echo/ApplicationInputHandler.cpp (hunk 4 above)
Comment (abstractmachines) [inline Echo/ApplicationInputHandler.cpp:1]: Yes :) Addressed here: https://github.com/PolySync/PolySync-CPP-Examples/pull/8/commits/99d437267e1d54b97c453efa9ce56ed7d566ed11

File: Echo/Echo.cpp (hunk 5 above)
Comment (abstractmachines) [inline Echo/Echo.cpp]: Addressed here: https://github.com/PolySync/PolySync-CPP-Examples/pull/8/commits/99d437267e1d54b97c453efa9ce56ed7d566ed11

**Pull request merged**